examples/factoranalysis/htfa_cv_example.py

```diff
@@ -0,0 +1,322 @@
+        HTFA weight matrix.
+   
+
+    Returns
+    -------
+    mse 
+        Returns mean square reconstruction error.
+
+    """
+
+    recon = F.dot(W).ravel()
+    err = mean_squared_error(
+        data.ravel(),
+        recon,
+        multioutput='uniform_average')
+    return math.sqrt(err)
+
+def get_train_err(htfa, data, F):
+    """Calcuate training error
+
+    Parameters
+    ----------
+
+    htfa : HTFA 
+        Factor Anaysis class in Brainiak.
+   
+    data : 2D array 
+        Input data to HTFA.
+
+    F : 2D array 
+        HTFA factor matrix.
+        
+    Returns
+    -------
+    mse 
+        Returns mean square error on training.
+
+    """
+
+    W = htfa.get_weights(data, F)
+    return recon_err(data, F, W)
+
+def get_test_err(htfa, test_weight_data, test_recon_data,
+        test_weight_R, test_recon_R, centers, widths):
+
+    """Calcuate test error
+    
+    Parameters
+    ----------
+
+    htfa : HTFA 
+        Factor Anaysis class in Brainiak.
+   
+    test_weigth_data : 2D array 
+        Data used for testing weights.
+    
+    test_recon_data : 2D array 
+        Data used for testing reconstruction error.
+
+    test_weigth_R : 2D array 
+        Coordinate matrix used for testing weights.
+    
+    test_recon_R : 2D array 
+        Coordinate matrix used for testing reconstruction error.
+
+    centers : 2D array 
+        Center matrix of HTFA factors.
+    
+    widths : 1D array 
+        Width matrix of HTFA factors.
+        
+    Returns
+    -------
+    mse 
+        Returns mean square error on test.
+
+    """
+
+    # calculate F on test_weight_R, based on trained centers/widths
+    unique_R, inds = htfa.get_unique_R(test_weight_R)
+    F = htfa.get_factors(unique_R,
+                         inds,
+                         centers,
+                         widths)
+    # calculate weights on test_weight_data
+    W = htfa.get_weights(test_weight_data, F)
+
+    # calculate F on final test_recon_data
+    unique_R, inds = htfa.get_unique_R(test_recon_R)
+    F = htfa.get_factors(unique_R,
+                         inds,
+                         centers,
+                         widths)
+    return recon_err(test_recon_data, F, W)
+
+
+comm = MPI.COMM_WORLD
+rank = comm.Get_rank()
+size = comm.Get_size()
+if rank == 0:
+    import logging
+    logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+n_subj = 2
+data_dir = os.path.join(os.getcwd(), 'data')
+if rank == 0 and not os.path.exists(data_dir):
+    os.makedirs(data_dir)
+
+url = []
+url.append(' https://www.dropbox.com/s/r5s9tg4ekxzbrco/s0.mat?dl=0')
+url.append(' https://www.dropbox.com/s/39tr01m76vxwaqa/s1.mat?dl=0')
+
+#get fMRI data and scanner RAS coordinates
+data = []
+R = []
+mapping = {}
+n_local_subj = 0
+for idx in range(n_subj):
+    if idx % size == rank:
+        mapping[str(n_local_subj)] = idx
+        n_local_subj += 1
+        file_name = os.path.join(data_dir, 's' + str(idx) + '.mat')
```

Comment:
I actually meant something much simpler: check if the file exist in the filesystem so the user can re-run the example without re-downloading the files every time.